publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/batch/dynamodb/LoadDynamodbResourceBatchJobHandler.java

```diff
@@ -0,0 +1,226 @@
+             new Environment().readEnv(PARALLEL_SQS_THREADS_ENV));
+    }
+
+    public LoadDynamodbResourceBatchJobHandler(AmazonDynamoDB dynamoDbClient,
+                                               SqsClient sqsClient,
+                                               EventBridgeClient eventBridgeClient,
+                                               String tableName,
+                                               String queueUrl,
+                                               String processingEnabled,
+                                               String parallelThreads) {
+        super(LoadDynamodbRequest.class);
+        this.dynamoDbClient = dynamoDbClient;
+        this.sqsClient = sqsClient;
+        this.eventBridgeClient = eventBridgeClient;
+        this.forkJoinPool = new ForkJoinPool(parseThreadCount(parallelThreads));
+        this.tableName = tableName;
+        this.queueUrl = queueUrl;
+        this.processingEnabled = processingEnabled;
+    }
+    
+    private static int parseThreadCount(String parallelThreads) {
+        return Optional.ofNullable(parallelThreads)
+            .filter(s -> !s.isBlank())
+            .map(Integer::parseInt)
+            .orElse(10);
+    }
+
+    @Override
+    protected LoadDynamodbResponse processInput(LoadDynamodbRequest input,
+                                                AwsEventBridgeEvent<LoadDynamodbRequest> event,
+                                                Context context) {
+        validateInput(input);
+
+        if (!isProcessingEnabled()) {
+            logger.warn("Processing is disabled via {} environment variable. Stopping execution.", 
+                       PROCESSING_ENABLED_ENV);
+            return new LoadDynamodbResponse(0, 0, input.getJobType());
+        }
+
+        logger.info("Processing batch with job type: {}, starting from marker: {}",
+                    input.getJobType(), input.getStartMarker());
+
+        var scanResult = dynamoDbClient.scan(getScanRequest(input.getStartMarker()));
+
+        var workItems = scanResult.getItems().stream()
+            .map(item -> {
+                var dynamoDbKey = new DynamodbResourceBatchDynamoDbKey(
+                    item.get(PRIMARY_KEY_PARTITION_KEY_NAME).getS(),
+                    item.get(PRIMARY_KEY_SORT_KEY_NAME).getS());
+                return new BatchWorkItem(dynamoDbKey, input.getJobType());
+            }).toList();
+
+        var batches = Lists.partition(workItems, SQS_BATCH_SIZE);
+        
+        var messagesQueued = CompletableFuture
+            .supplyAsync(() -> batches.parallelStream()
+                .mapToInt(this::sendBatchToQueue)
+                .sum(), forkJoinPool)
+            .exceptionally(e -> {
+                logger.error("Failed to process batches in parallel, falling back to sequential", e);
+                return batches.stream()
+                    .mapToInt(this::sendBatchToQueue)
+                    .sum();
+            })
+            .join();
```

Comment:
What about trying parallel scan here? In order to make it way faster?

. Use Parallel Scans for Large Tables

**DynamoDB allows you to divide a scan into multiple segments that run concurrently**
Specify TotalSegments and Segment parameters to distribute the workload
This is most effective when you need to process the entire table quickly
Example: Split into 4-8 segments depending on table size and provisioned throughput